player.go

```diff
@@ -3,12 +3,16 @@ package spotify
 import (
 	"bytes"
 	"encoding/json"
+	"errors"
 	"net/http"
 	"net/url"
 	"strconv"
 	"time"
 )
 
+// ErrNoPlayer is an error returned when spotify could not find a connected player
+var ErrNoPlayer = errors.New("spotify: no player connected")
```

Comment:
I'm glad we're not returning an EOF anymore, but I'm not sure I like the idea of returning an error for a 204 status code.

204 is a successful code.  Spotify does not see this as an error, so why should this API?

Reply:
Hmm true, I was thinking of making it easy for the user to check no player was returned.
You'r rather want to return Nil or an empty struct and let users figure out themselves it was and empty response?